src/Injector.ts

```diff
@@ -108,37 +108,32 @@ class Injector {
       return false;
     };
 
+    let rules;
     if (/channels/.test(inputPath)) {
-      // Exclude paths
-      if (this.globCheck(operationName, injectRule.excludeChannels, picomatchOptions)) {
-        return false;
-      }
-      if (this.globCheck(operationName, injectRule.exclude, picomatchOptions)) {
-        return false;
-      }
-      // Specifically include a path
-      if (injectRule.includeOnlyChannels.length > 0 && !this.globCheck(operationName, injectRule.includeOnlyChannels, picomatchOptions)) {
-        return false;
-      }
-      // include method
-      if (shouldSkipMethod(methodName)) {
-        return false;
-      }
+      rules = { exclude: injectRule.exclude, route: injectRule.excludeChannels, only: injectRule.includeOnlyChannels };
+    } else if (/paths/.test(inputPath)) {
+      rules = { route: injectRule.excludePaths, only: injectRule.includeOnlyPaths };
+    } else {
+      return true;
     }
-    if (/paths/.test(inputPath)) {
-      // Exclude a path completely
-      if (this.globCheck(operationName, injectRule.excludePaths, picomatchOptions)) {
-        return false;
-      }
-      // Specifically include a path
-      if (injectRule.includeOnlyPaths.length > 0 && !this.globCheck(operationName, injectRule.includeOnlyPaths, picomatchOptions)) {
-        return false;
-      }
-      // include method
-      if (shouldSkipMethod(methodName)) {
-        return false;
-      }
+
+    // Exclude paths
+    if (this.globCheck(operationName, rules.route, picomatchOptions)) {
+      return false;
+    }
+    // Exclude (channel only)
+    if (rules.exclude && this.globCheck(operationName, rules.exclude, picomatchOptions)) {
+      return false;
+    }
+    // Specifically include a path
+    if (rules.only?.length > 0 && !this.globCheck(operationName, rules.only, picomatchOptions)) {
```

Comment:
is this line correct? looks like it should be includeOnlyPaths not only

Reply:
Yeah, I just renamed them above:
```
rules.exclude = injectRule.exclude (channel only)
rules.route = injectRule.excludeChannels | injectRule.excludePaths
rules.only = injectRule.includeOnlyChannels | injectRule.includeOnlyPaths
```

Since `includeOnlyChannels` = `includeOnlyPaths`, and `excludeChannels` = `excludePaths` (for channels vs paths), it's really doing the same check on channels and paths, except with different inject keys, with the exception of `injectRule.exclude` which is a channel-specific injection setting.

I'm not 100% sure what all the channel settings are for, but we could probably streamline configurations here where regardless of the api style, the injection include/exclude logic would probably just be whitelist | blacklist | regex

Reply:
I just renamed them again as those names were not good:
```
rules.exclude = injectRule.excludeChannels | injectRule.excludePaths
rules.include = injectRule.includeOnlyChannels | injectRule.includeOnlyPaths
injectRule.exclude is just checked as-is
```